youtube/yt_data_extract/watch_extraction.py

```diff
@@ -752,6 +752,27 @@ def extract_watch_info_from_html(watch_html):
 def captions_available(info):
     return bool(info['_captions_base_url'])
 
+def parse_endcard(card):
+    """
+    parses a single endcard into a format that's easier to handle.
+    from: https://git.gir.st/subscriptionfeed.git/blob/737a2f6f:/app/common/innertube.py#l301
+    """
+    card = card.get('endscreenElementRenderer', card) #only sometimes nested
+    ctype = card['style'].lower()
+    if ctype == "video":
+        if not 'endpoint' in card: return None # title == "This video is unavailable."
+        video_id = card['endpoint']['watchEndpoint']['videoId']
+        return {'type': ctype,
+            'video_id': video_id,
+            'title': extract_str(card['title']),
+            'approx_view_count': extract_str(card['metadata']),
+            'thumbnail': f"/https://i.ytimg.com/vi/{video_id}/default.jpg",
+            'duration': extract_str(card["thumbnailOverlays"][0]["thumbnailOverlayTimeStatusRenderer"]["text"])
+            # XXX: no channel name
+        }
```

Comment:
This code needs to use the deep_get function. Never extract information from json like this: `card['endpoint']['watchEndpoint']['videoId']`, that will cause the entire code to fail when YouTube changes their json.

Also some lines need to be split up onto multiple lines to not exceed 79 character line limit. And use ', not ", for strings. See https://github.com/user234683/youtube-local/blob/master/HACKING.md for coding guidelines